test/generators/java/presets/__snapshots__/JacksonPreset.spec.ts.snap

```diff
@@ -64,10 +64,10 @@ Array [
  * Vehicle represents a union of types: Car, Truck
  */
 public interface Vehicle {
-  String getVehicleType();
```

Comment:
Was dropping this intentional? The current functionality exposes the discriminator if it is the same type across all the union types.

Reply:
I suspect this is caused by a similar name matching issue.

Reply:
Good catch. The same fix was needed in `addDiscriminatorTypeToUnionModel` which I just added.